eventlet/hubs/__init__.py

```diff
@@ -26,43 +34,28 @@ def get_default_hub():
     .. include:: ../doc/common.txt
     .. note :: |internal|
     """
+    for mod in builtin_hub_modules:
+        if mod.is_available():
+            return mod
 
-    # pyevent hub disabled for now because it is not thread-safe
-    # try:
-    #    import eventlet.hubs.pyevent
-    #    return eventlet.hubs.pyevent
-    # except:
-    #    pass
-
-    select = patcher.original('select')
-    try:
-        import eventlet.hubs.epolls
-        return eventlet.hubs.epolls
-    except ImportError:
-        try:
-            import eventlet.hubs.kqueue
-            return eventlet.hubs.kqueue
-        except ImportError:
-            if hasattr(select, 'poll'):
-                import eventlet.hubs.poll
-                return eventlet.hubs.poll
-            else:
-                import eventlet.hubs.selects
-                return eventlet.hubs.selects
+    raise Exception('all built-in hubs are not available')
 
 
 def use_hub(mod=None):
     """Use the module *mod*, containing a class called Hub, as the
     event hub. Usually not required; the default hub is usually fine.
 
-    Mod can be an actual module, a string, or None.  If *mod* is a module,
-    it uses it directly.   If *mod* is a string and contains either '.' or ':'
-    use_hub tries to import the hub using the 'package.subpackage.module:Class'
-    convention, otherwise use_hub looks for a matching setuptools entry point
-    in the 'eventlet.hubs' group to load or finally tries to import
-    `eventlet.hubs.mod` and use that as the hub module.  If *mod* is None,
-    use_hub uses the default hub.  Only call use_hub during application
-    initialization,  because it resets the hub's state and any existing
+    `mod` can be an actual hub class, a module, a string, or None.
+
+    If `mod` is a class, use it directly.
+    If `mod` is a module, use `module.Hub` class
+    If `mod` is a string and contains either '.' or ':'
+    then uses 'package.subpackage.module:Class' convention,
+    otherwise imports `eventlet.hubs.mod`.
+    If `mod` is None, `use_hub` uses the default hub.
```

Comment:
I see tests for passing `mod` as a class, and for checking the default hub with `kqueue` hidden. Do we want to add tests for passing `mod` as a module, `mod` as a string without punctuation, `mod` as a string with punctuation, `mod` as None?

Reply:
The more tests the merrier! I included the least required amount of tests to cover changes. For lazy reasons. Is it a merge blocker?